### PR TITLE
Improve segmentation

### PR DESCRIPTION
**Pull request by GFabien**

### CODE REVIEW

File: segmentation/config.py [hunk 1]
```diff
@@ -387,22 +387,28 @@ def create_dataset(subjects, transforms):
...
-            train_loader = DataLoader(train_queue, self.batch_size, collate_fn=struct['collate_fn'])
-            val_queue = torchio.Queue(self.val_set, **struct['queue']['attributes'])
-            val_loader = DataLoader(val_queue, self.batch_size, collate_fn=struct['collate_fn'])
+            if len(self.train_set) > 0:
+                train_queue = torchio.Queue(self.train_set, **struct['queue']['attributes'])
+                train_loader = DataLoader(train_queue, self.batch_size, collate_fn=struct['collate_fn'])
+            if len(self.val_set) > 0:
+                val_queue = torchio.Queue(self.val_set, **struct['queue']['attributes'])
+                val_loader = DataLoader(val_queue, self.batch_size, collate_fn=struct['collate_fn'])
         return train_loader, val_loader
```
Comment (romainVala) [inline segmentation/config.py:415]: du cou on peut aussi avoir un validation null (a tester aussi si ca passe partour)

File: segmentation/config.py [hunk 2]
```diff
@@ -373,6 +372,10 @@ def check_modalities(ref_modalities, modalities, subject_name):
         val_subjects += subjects[end_train:end_val]
         test_subjects += subjects[end_val:]
 
+        self.logger.log(logging.INFO, f'{len(train_subjects)} subjects in the train set')
+        self.logger.log(logging.INFO, f'{len(val_subjects)} subjects in the validation set')
+        self.logger.log(logging.INFO, f'{len(test_subjects)} subjects in the test set')
+
         return train_subjects, val_subjects, test_subjects
```
Comment (romainVala) [inline segmentation/config.py:379]: Add a line to specify shuffle and seeding
"Loading subject, whith shuffle=True seeding is None "
Can you add the path ( the path ) of the first subject with the modality name ?
for instance,
Train set : x subject the first one has /path1 for modality T1 /path2 for modality label ect ...

File: segmentation/run_model.py [hunk 3]
```diff
@@ -67,8 +67,6 @@ def __init__(self, model, train_loader, val_loader, val_set, test_set, image_key
         # Set attributes to keep track of information during training
         self.epoch = struct['current_epoch']
         self.iteration = 0
-        self.train_df = None
-        self.val_df = None
```
Comment (romainVala) [inline segmentation/run_model.py:70]: cool, c'est plus "propre" sans avoir a dissocier val et train

File: segmentation/run_model.py [hunk 4]
```diff
@@ -210,13 +208,10 @@ def train_loop(self, save_model=True):
...
-            if self.model.training and i % self.record_frequency == 0:
-                self.batch_recorder(sample, predictions, targets, batch_time, True)
-                self.train_df = pd.DataFrame()
-            elif self.model.training:
-                self.train_df = self.batch_recorder(sample, predictions, targets, batch_time, False)
+            if i % self.record_frequency == 0 or i == len(loader):
+                df = self.batch_recorder(df, sample, predictions, targets, batch_time, True)
             else:
-                self.val_df = self.batch_recorder(sample, predictions, targets, batch_time, True)
+                df = self.batch_recorder(df, sample, predictions, targets, batch_time, False)
```
Comment (romainVala) [inline segmentation/run_model.py:215]: je pensais que pour la 2ieme epoch il faudrait faire, l'initialisation, (df = pd.DataFrame() ) ... mais je me trompe surement
